app/gui/src/controller/searcher/component.rs

```diff
@@ -4,92 +4,135 @@
 
 use crate::prelude::*;
 
+use crate::controller::graph::RequiredImport;
+use crate::controller::searcher::Filter;
 use crate::model::suggestion_database;
 
-use controller::searcher::action::MatchKind;
-use controller::searcher::Filter;
-use convert_case::Case;
-use convert_case::Casing;
-use double_representation::name::QualifiedName;
+use double_representation::name::project;
 use enso_doc_parser::DocSection;
 use enso_doc_parser::Tag;
+use enso_suggestion_database::entry;
+use ensogl::data::color;
 use ordered_float::OrderedFloat;
+use std::cmp;
+use superslice::Ext;
 
 
 // ==============
 // === Export ===
 // ==============
 
 pub mod builder;
-pub mod group;
 pub mod hardcoded;
 
-pub use group::Group;
+pub use builder::Builder;
 
 
 
 // =================
 // === Constants ===
 // =================
 
-/// A "matching" score assigned to the entries which does not match the current pattern entirely.
-///
-/// **Note**: If some entries matches, but their score are equal or below this value, they will be
-/// filtered out as well!
-pub const NOT_MATCHING_SCORE: f32 = 0.0;
-
 /// A factor to multiply a component's alias match score by. It is intended to reduce the importance
 /// of alias matches compared to label matches.
 const ALIAS_MATCH_ATTENUATION_FACTOR: f32 = 0.75;
 
 
 
-// ====================
-// === Type Aliases ===
-// ====================
+// =============
+// === Group ===
+// =============
 
-/// A component identifier.
-pub type Id = suggestion_database::entry::Id;
-/// Information how the component matches the filtering pattern.
-pub type MatchInfo = controller::searcher::action::MatchInfo;
+/// A structure describing a component group.
```

Comment:
Maybe just "A group of components"? Every `struct` is a structure.

Reply:
Technically, the structure does not contain components, so I rephrased it differently.